controllers/nodefeaturediscovery_controls.go

```diff
@@ -305,6 +309,15 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
 		obj.Spec.Template.Spec.Containers[0].ImagePullPolicy = n.ins.Spec.Operand.ImagePolicy(n.ins.Spec.Operand.ImagePullPolicy)
 	}
 
+	// update nfd-master service port
+	if n.ins.Spec.Operand.ServicePort != 0 && obj.ObjectMeta.Name == "nfd-master" {
+		portFlag := fmt.Sprintf("--port=%d", n.ins.Spec.Operand.ServicePort)
+		obj.Spec.Template.Spec.Containers[0].Args = []string{portFlag}
+	} else if obj.ObjectMeta.Name == "nfd-master" {
+		portFlag := fmt.Sprintf("--port=%d", defaultServicePort)
+		obj.Spec.Template.Spec.Containers[0].Args = []string{portFlag}
+	}
```

Comment:
Hmm, my comment regarding this was lost somewhere yesterday 🤔  Anyway, I somehow find this more difficult to read than it should. What do you think about something like this (could probably be made polished even more but the main point is checking for "nfd-master" only once)

```suggestion
        if obj.ObjectMeta.Name == "nfd-master" {
                port := defaultServicePort
                if n.ins.Spec.Operand.ServicePort != 0 {
                        port = n.ins.Spec.Operand.ServicePort
                }
                portFlag := fmt.Sprintf("--port=%d", port)
                obj.Spec.Template.Spec.Containers[0].Args = []string{portFlag}
        }

```